README.md

```diff
@@ -33,25 +86,169 @@ VS Code will generate a `launch.json` in your project with some default configur
 
-See [Setting up debug environment](doc/debugging.md) for more details.
+## Expo applications
 
-## React Native debug configuration properties
+To debug a project created using Expo or the create-react-native-app task, you can use embedded support for Expo.
+
+Your environment must meet the following prerequisites:
+
+- Install the [Expo app](https://getexponent.com/) on the target device or emulator
+- Ensure that the react-native-cli is installed globally (`npm install -g react-native-cli`)
+
+To start debugging in Expo follow these steps:
+
+1. Open your project in VS Code with this extension installed.
+1. Create a debug configuration (as described in [Debugging React Native applications](#debugging-react-native-applications)), select `Debug in Exponent` in the debug drop-down menu, and start debugging
+1. Wait while some dependencies are configured - the extension will install `xde` and `@expo/ngrok` when this feature is used for the first time.
+1. If you have not used exponent on this system before, you will be prompted for an exponent username and password.
+   If you have not created an exponent account, then specifying a new username and password will create one.
+   Note that there is no e-mail associated with the account, and no way to recover a forgotten password.
+1. Once the packager starts, the extension will open a separate tab with QR code to scan from the Exponent app. Once you do so, the Exponent app will connect to the packager and begin running your app.
+1. Once the app is loaded and running, [open the developer menu](https://reactnative.dev/docs/debugging#accessing-the-in-app-developer-menu) and enable remote debugging by clicking on `Debug JS Remotely` button.
+
+   ![React Native developer menu](./images/enable-remote-debug.png)
+
+   From here you can run and debug the app as normal.
+
+### Configuring Expo
+
+The extension supports running through Exponent not just the applications created with Expo but even pure React Native applications (in that case you need to add `expo` package to `node_modules` in order to make it work with Expo: `npm install expo --save-dev`. In either cases it uses `app.json` configuration file in the root of the project.
+
+If you are running `Debug in Exponent` configuration or any of pallette commands like `Run in Exponent`, `Publish to Exponent` then this file will be created automatically if absent or updated with the following basic configuration section:
+
+```json
+{
+  "expo": {
+    "slug": "MyApp", // Project slug
+    "name": "MyApp", // Project name
+    "sdkVersion": "31.0.0", // Expo SDK version
+    "entryPoint": ".vscode\\exponentIndex.js" // Entrypoint for the project
+  },
+  "name": "MyApp" // Project name
+}
+```
+
+Full list of configurational parameters for `expo` section in `app.json` may be found on [official Expo documentation page](https://docs.expo.io/versions/latest/workflow/configuration).
```

Comment:
should this be configuration over configurational?

Reply:
> should this be configuration over configurational?

configuration sounds better to me but English is not my first language so 🤷‍♂️

```suggestion
Full list of configuration parameters for `expo` section in `app.json` may be found on [official Expo documentation page](https://docs.expo.io/versions/latest/workflow/configuration).
```